homeassistant/components/shelly/cover.py

```diff
@@ -0,0 +1,109 @@
+"""Light for Shelly."""
+from aioshelly import Block
+
+from homeassistant.components.cover import (
+    CoverEntity,
+    SUPPORT_OPEN,
+    SUPPORT_CLOSE,
+    SUPPORT_STOP,
+    SUPPORT_SET_POSITION,
+)
+from homeassistant.const import (
+    SERVICE_CLOSE_COVER,
+    SERVICE_OPEN_COVER,
+    SERVICE_SET_COVER_POSITION,
+    SERVICE_STOP_COVER,
+    STATE_CLOSED,
+    STATE_CLOSING,
+    STATE_OPEN,
+    STATE_OPENING,
+)
+from homeassistant.core import callback
+
+from . import ShellyBlockEntity, ShellyDeviceWrapper
+from .const import DOMAIN
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up cover for device."""
+    wrapper = hass.data[DOMAIN][config_entry.entry_id]
+    blocks = [block for block in wrapper.device.blocks if block.type == "roller"]
+
+    if not blocks:
+        return
+
+    async_add_entities(ShellyCover(wrapper, block) for block in blocks)
+
+
+class ShellyCover(ShellyBlockEntity, CoverEntity):
+    """Switch that controls a cover block on Shelly devices."""
+
+    def __init__(self, wrapper: ShellyDeviceWrapper, block: Block) -> None:
+        """Initialize light."""
+        super().__init__(wrapper, block)
+        self.control_result = None
+        self._supported_features = 0
+
+    @property
+    def is_closed(self):
+
+        return self.block.rollerPos == 0
+
+    @property
+    def current_cover_position(self):
+        """Position of the cover."""
+        if self.control_result:
+            cover_postion = self.control_result["rollerPos"]
+        else:
+            cover_position = self.block.rollerPos
+        return cover_position
+
+    @property
+    def should_poll(self):
+        """No polling needed."""
+        return True
+
+    @property
+    def is_closing(self):
+        """Return if the cover is closing."""
+        return self.block.roller == "close"
+
+    @property
+    def is_opening(self):
+        """Return if the cover is opening."""
+        return self.block.roller == "open"
```

Comment:
`is_opening` and `is_closing` check if the device is in the process of opening/closing. YOu just want to implement `is_closed` 
```suggestion
    @property
    def is_closed(self):
        """Return if the cover is closing."""
        return self.block.roller == "close"
```

Reply:
Currently ShellyForHASS is implementing also the 2 process. So I though to implement the same features ;-)

Reply:
What does `state` represents ? Does it represent the current action or the state of the roller shutter.

Reply:
And is this something that we should have the `control_result` override ?

Reply:
> What does `state` represents ? Does it represent the current action or the state of the roller shutter.

Current action, this is why it works fine with those 2 methods in HA